tests/fsharp/test-framework.fs

```diff
@@ -156,34 +142,36 @@ let config configurationName envVars =
 
     let SCRIPT_ROOT = __SOURCE_DIRECTORY__
     let packagesDir = Environment.GetEnvironmentVariable("USERPROFILE") ++ ".nuget" ++ "packages"
-#if NET472
+#if INCLUDE_NETFX_TESTS
     let fscArchitecture = "net472"
     let fsiArchitecture = "net472"
     let fsharpCoreArchitecture = "net45"
     let fsharpBuildArchitecture = "net472"
     let fsharpCompilerInteractiveSettingsArchitecture = "net472"
-#else
-    let fscArchitecture = "netcoreapp2.1"
```

Comment:
Won't this force the wrong `fsc.exe` to be invoked during a test by always using the netfx one?

Reply:
There's important things here 

1. The testdriver is really one big testing script/application

2. The framework being used to run the testdriver (which is now always netcoreapp2.1) is not the framework being tested.

3. The test driver only ever invokes `fsc.exe` (or `ildasm` or `csc` or `peverify` or a bunch of other things) directly when running .NET Framework tests.  In these cases it runs the desktop compiler/tools when using direct compiler invocations (as before).